### ISSUE TITLE
OneCycle annealing

### ISSUE THREAD

**Issue by MilesCranmer** [FluxML/ParameterSchedulers.jl#42]
### Describe the potential feature

One cycle annealing (original paper [here](https://arxiv.org/abs/1708.07120)) is a really strong scheduler and what I've found to be the most optimal scheduler for deep learning. An implementation in PyTorch is here: https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.OneCycleLR.html

### Possible Implementation

```julia
"Cosine anneal from `start` to `end` as pct goes from 0.0 to 1.0."
function annealing_cos(start, stop, pct)
    cos_out = cos(pi * pct) + 1
    return stop + (start - stop) / 2.0 * cos_out
end
```

You would typically start at pct 0.3, and go to pct=1.0 at the end of training.

**Comment by MilesCranmer** [comment]
(You could probably just build off of `CosAnneal`, with `period=nsteps`, and have it start 30% way through the cycle)

**Comment by darsnack** [comment]
There is an implementation [here](https://github.com/FluxML/FluxTraining.jl/blob/73b8d0b8d961f3423d3727f0b42491ddba8b314b/src/callbacks/scheduler.jl#L68-L93). We can adapt that into a convenience constructor in ParameterSchedulers.jl.

**Comment by MilesCranmer** [comment]
Cool! That sounds perfect.

**Comment by MilesCranmer** [comment]
Just pinging this, let me know if this is possible

**Comment by darsnack** [comment]
Yes, definitely do-able. Just need to do it. A PR doing it before I do would be appreciated and accepted.
